Remove the #if test for HAVE_DECL_BASENAME and just assume that
basename(3) is always defined.  Basename is POSIX 2001, so it will
surely exist.  Without this, the binutils declaration for basename
conflicts with the one in string.h.

This is a consequence of binutils (incorrectly) assuming that
HAVE_DECL_BASENAME has been set from config.h.


diff -Naurb binutils-2.26.orig/include/libiberty.h binutils-2.26/include/libiberty.h
--- binutils-2.26.orig/include/libiberty.h	2015-11-13 02:27:42.000000000 -0600
+++ binutils-2.26/include/libiberty.h	2016-03-03 18:21:33.054230365 -0600
@@ -104,7 +104,7 @@
    declaration without arguments.  If it is 0, we checked and failed
    to find the declaration so provide a fully prototyped one.  If it
    is 1, we found it so don't provide any declaration at all.  */
-#if !HAVE_DECL_BASENAME
+#if 0
 #if defined (__GNU_LIBRARY__ ) || defined (__linux__) \
  || defined (__FreeBSD__) || defined (__OpenBSD__) || defined (__NetBSD__) \
  || defined (__CYGWIN__) || defined (__CYGWIN32__) || defined (__MINGW32__) \
